lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1282,6 +1282,41 @@ void SPIRVToLLVM::addMemAliasMetadata(Instruction *I, SPIRVId AliasListId,
   I->setMetadata(AliasMDKind, MDAliasListMap[AliasListId]);
 }
 
+void transFunctionPointerCallArgumentAttributes(SPIRVValue *BV, CallInst *CI) {
+  std::vector<SPIRVDecorate const *> ArgumentAttributes =
+      BV->getDecorations(internal::DecorationArgumentAttributeINTEL);
+
+  for (const auto *Dec : ArgumentAttributes) {
+    std::vector<SPIRVWord> Literals = Dec->getVecLiteral();
+    SPIRVWord ArgNo = Literals[0];
+    SPIRVWord Attr = Literals[1];
+
+    AttrBuilder Builder;
+    if (Attr == FunctionParameterAttributeByVal)
```

Comment:
If only one attr is expected per decoration then it would be better to use `switch`.

Reply:
That's right, one attribute per decoration.